pkg/cloud/services/eks/nodegroup.go

```diff
@@ -599,3 +648,19 @@ func (s *NodegroupService) waitForNodegroupActive() (*eks.Nodegroup, error) {
 
 	return ng, nil
 }
+
+// markReadyAndUpdateConditionsTrue sets the EKSNodegroupReadyCondition to true, and if an update was in progress,
+// it also sets the EKSNodegroupUpdateSucceededCondition to true.
+func (s *NodegroupService) markReadyAndUpdateConditionsTrue() {
+	conditions.MarkTrue(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupReadyCondition)
+	if s.isNodegroupConditionUpdating() {
+		conditions.MarkTrue(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition)
+	}
+}
+
+// isNodegroupConditionUpdating checks if the node group is currently in an updating condition (i.e. an update
+// was previously initiated).
+func (s *NodegroupService) isNodegroupConditionUpdating() bool {
+	return conditions.IsFalse(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition) &&
```

Comment:
I'm no in favour of using conditions for logic control. If a not group is updating we should be able to get this from the status of the nodegroup from AWS.

Reply:
@jon-fearer - just checking in on this to see what you though?